pkg/plugin/plugin.go

```diff
@@ -0,0 +1,253 @@
+		if !strings.HasPrefix(name, "limactl-") {
+			continue
+		}
+
+		pluginName := strings.TrimPrefix(name, "limactl-")
+
+		if strings.Contains(pluginName, ".") {
+			ext := filepath.Ext(name)
+			if !isWindowsExecutableExt(ext) {
+				continue
+			}
+
+			pluginName = strings.TrimSuffix(pluginName, ext)
+		}
+
+		fullPath := filepath.Join(dir, name)
+
+		if !isExecutable(fullPath) {
+			continue
+		}
+
+		plugin := Plugin{
+			Name: pluginName,
+			Path: fullPath,
+		}
+
+		if desc := extractDescFromScript(fullPath); desc != "" {
+			plugin.Description = desc
+		}
+
+		plugins = append(plugins, plugin)
+	}
+
+	return plugins
+}
+
+func isExecutable(path string) bool {
+	info, err := os.Stat(path)
+	if err != nil {
+		return false
+	}
+
+	mode := info.Mode()
+	if mode&0o111 != 0 {
+		return true
+	}
+
+	if runtime.GOOS == "windows" {
+		ext := filepath.Ext(path)
+		return isWindowsExecutableExt(ext)
+	}
+
+	return false
+}
+
+func RunExternalPlugin(ctx context.Context, name string, args []string) {
+	if ctx == nil {
+		ctx = context.Background()
+	}
+
+	if err := UpdatePathForPlugins(); err != nil {
+		logrus.Warnf("failed to update PATH environment: %v", err)
+		// PATH update failure shouldn't prevent plugin execution
+	}
+
+	plugins, err := DiscoverPlugins()
+	if err != nil {
+		logrus.Warnf("failed to discover plugins: %v", err)
+		return
+	}
+
+	var execPath string
+	for _, plugin := range plugins {
+		if plugin.Name == name {
+			execPath = plugin.Path
+			break
+		}
+	}
+
+	if execPath == "" {
+		return
+	}
+
+	cmd := exec.CommandContext(ctx, execPath, args...)
+	cmd.Stdin = os.Stdin
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	cmd.Env = os.Environ()
+
+	err = cmd.Run()
+	usrlocalsharelima.HandleExitError(err)
+	if err == nil {
+		os.Exit(0) //nolint:revive // it's intentional to call os.Exit in this function
+	}
+	logrus.Fatalf("external command %q failed: %v", execPath, err)
+}
+
+var descRegex = regexp.MustCompile(`<limactl-desc>(.*?)</limactl-desc>`)
+
+func extractDescFromScript(path string) string {
+	content, err := os.ReadFile(path)
+	if err != nil {
+		logrus.Debugf("Failed to read plugin script %s: %v", path, err)
+		return ""
+	}
+
+	if !strings.HasPrefix(string(content), "#!") {
+		logrus.Debugf("Plugin %s: not a script file, skipping description extraction", filepath.Base(path))
+		return ""
+	}
+
+	matches := descRegex.FindAllStringSubmatch(string(content), -1)
+	if len(matches) == 0 {
+		logrus.Debugf("Plugin %s: no <limactl-desc> found in script", filepath.Base(path))
+		return ""
+	}
+
+	var descs []string
+	for _, match := range matches {
+		if len(match) > 1 {
+			descs = append(descs, strings.TrimSpace(match[1]))
+		}
+	}
+
+	if len(descs) == 0 {
+		return ""
+	}
+
+	mergedDesc := descs[0]
+	for _, line := range descs[1:] {
+		mergedDesc += "\n" + strings.Repeat(" ", 22) + line
+	}
+
+	logrus.Debugf("Plugin %s: extracted merged description from script:\n%q", filepath.Base(path), mergedDesc)
+	return mergedDesc
```

Comment:
I think you should ignore everything but the first match. And you definitely should not include multi-line formatting in the description here; that would have to be done at the presentation layer. But I think we should simply recommend to use short, one-line descriptions.

So the whole block should be just `return matches[0]` and that's it.